checks/apps/quantumespresso/quantumespresso_check.py

```diff
@@ -48,106 +91,44 @@ def time(self):
-                self.num_cpus_per_task = 16
-                self.num_tasks_per_core = 1
-                self.env_vars = {
-                    'MPICH_OFI_STARTUP_CONNECT': 1,
-                    'OMP_NUM_THREADS': 8,
-                    'OMP_PLACES': 'cores',
-                    'OMP_PROC_BIND': 'close'
-                }
-
     @run_before('performance')
-    def set_reference(self):
-        references = {
-            'small': {
-                'dom:mc': {'time': (109.814, None, 0.10, 's')},
-                'daint:mc': {'time': (98.475, None, 0.10, 's')},
-                'eiger:mc': {'time': (55.51, None, 0.10, 's')},
-                'pilatus:mc': {'time': (50.981, None, 0.10, 's')}
-            },
-            'large': {
-                'daint:mc': {'time': (125.648, None, 0.10, 's')},
-                'eiger:mc': {'time': (47.165, None, 0.10, 's')},
-                'pilatus:mc': {'time': (43.211, None, 0.10, 's')}
-            }
+    def set_performance_reference(self):
+        self.reference = {
+            '*': {'time': (self.ref_dict['performance_reference'][0]['P'],
+                           None, 0.10, 's')}
         }
-        self.reference = references[self.scale]
-
-    @run_before('run')
-    def set_task_distribution(self):
-        self.job.options = ['--distribution=block:block']
-
-    @run_before('run')
-    def set_cpu_binding(self):
-        self.job.launcher.options = ['--cpu-bind=cores']
 
 
 @rfm.simple_test
-class QuantumESPRESSOGpuCheck(QuantumESPRESSOCheck):
-    valid_systems = ['daint:gpu']
-    num_gpus_per_node = 1
-    num_tasks_per_node = 1
-    num_cpus_per_task = 12
-    energy_tolerance = 1.0e-7
+class UENV_QuantumESPRESSOCheck(QuantumESPRESSOBase,
+                                ExtraLauncherOptionsMixin,
+                                CudaVisibleDevicesAllMixin):
+    valid_systems = ['+uenv -amdgpu']
+    valid_prog_environs = ['+quantum-espresso', '+mpi', '+openmp']
+    tags = {'production', 'uenv'}
+    energy_tolerance = 1.0e-6
+    test_name = parameter(['Au-surf'])
 
-    @run_after('init')
-    def setup_test(self):
-        self.descr = (f'QuantumESPRESSO GPU check (version: {self.scale}, '
-                      f'{self.variant})')
-        if self.scale == 'small':
-            self.valid_systems += ['dom:gpu']
-            self.num_tasks = 6
-            self.energy_reference = -11427.09017168
+@rfm.simple_test
+class SARUS_QuantumESPRESSOCheck(QuantumESPRESSOBase,
+                                 SarusExtraLauncherOptionsMixin,
+                                 CudaVisibleDevicesAllMixin):
+    container_image = variable(str, value='NULL')
+    valid_prog_environs = ['builtin']
+    valid_systems = ['+remote']
+    test_name = parameter(['Au-surf'])
+    tags = {'production', 'sarus'}
+
+    @run_after('setup')
+    def setup_container_platform(self):
+        # if container_image was provided then initialize a container execution
+        if self.container_image != 'NULL':
```

Comment:
```suggestion
        if self.container_image:
```

Reply:
@teojgo If I change the `container_image = variable(str, type(None), value=None)`, this will be a vaild change here

Reply:
You should explicitly check for None in this case, because you might be able to set an empty string. So checking for None would make sure that nothing is passed from the command line

Reply:
I didn't get from Theo's comment what to do here.

Reply:
It's the same in both cases. In case you want to go with the version where the default is `None`, you better check that :
    if self.container_image is None: